src/Microsoft.OpenApi/Models/OpenApiHeader.cs

```diff
@@ -82,17 +77,17 @@ public class OpenApiHeader : IOpenApiSerializable, IOpenApiReferenceable, IOpenA
         /// <summary>
         /// Examples of the media type.
         /// </summary>
-        public IDictionary<string, OpenApiExample> Examples { get; set; } = new Dictionary<string, OpenApiExample>();
+        public IDictionary<string, OpenApiExample> Examples { get; set; }
```

Comment:
It was an intentional design choice to ensure collections were non-null so that library users did not have to check if collections had been created before adding properties to the collection.  However, due to the performance cost we should do a lazy instantiation of these properties in the get accessor instead of doing it at create time.

Reply:
These initializations cause these properties to serialize.  Lazy initialization will do the same.  Not using custom JSON converters is preferred, where possible.

For ease-of-use, personally, I think a fluent builder is better.